gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageWriteChannel.java

```diff
@@ -26,11 +26,12 @@
 import com.google.cloud.hadoop.util.ClientRequestHelper;
 import com.google.cloud.hadoop.util.LoggingMediaHttpUploaderProgressListener;
 import java.io.IOException;
+import java.io.InputStream;
+import java.util.concurrent.Callable;
 import java.util.concurrent.ExecutorService;
 
 /** Implements WritableByteChannel to provide write access to GCS. */
-public class GoogleCloudStorageWriteChannel
-    extends AbstractGoogleAsyncWriteChannel<Storage.Objects.Insert, StorageObject>
+public class GoogleCloudStorageWriteChannel extends AbstractGoogleAsyncWriteChannel<StorageObject>
```

Comment:
Should we consider collapsing this hierarchy as well ?

Reply:
No, `AbstractGoogleAsyncWriteChannel` used by JSON and gRPC write channels, so it's actually used/useful.